compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -1865,16 +1866,25 @@ public void visit(BLangPackageVarRef astPackageVarRefExpr) {
 
     private BIRGlobalVariableDcl getVarRef(BLangPackageVarRef astPackageVarRefExpr) {
         BSymbol symbol = astPackageVarRefExpr.symbol;
-        if ((symbol.tag & SymTag.CONSTANT) == SymTag.CONSTANT ||
-                !isInSamePackage(astPackageVarRefExpr.varSymbol, env.enclPkg.packageID)) {
-            return new BIRGlobalVariableDcl(astPackageVarRefExpr.pos, symbol.flags, symbol.type, symbol.pkgID,
-                                            symbol.name, symbol.getOriginalName(), VarScope.GLOBAL, VarKind.CONSTANT,
-                                            symbol.name.value, symbol.origin.toBIROrigin());
+        Location pos = astPackageVarRefExpr.pos;
+
+        if (!isInSamePackage(astPackageVarRefExpr.varSymbol, env.enclPkg.packageID)) {
+            BIRGlobalVariableDcl newGlobalVarDcl = getNewGlobalVarDcl(pos, symbol);
+            this.env.enclPkg.importedGlobalVars.add(newGlobalVarDcl);
```

Comment:
Since it is a temporary global variable we can still add to global vars instead of creating new list for imported global vars.